projects/pastanaga-angular/src/lib/modal/modal.model.ts

```diff
@@ -1,31 +1,43 @@
-import { EventEmitter } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
 
-export interface IModalConfig {
+export interface IModalConfig<D = any> {
     blocking?: boolean;
     withCloseButton?: boolean;
+    data?: D;
 }
 
-export class ModalConfig {
+export class ModalConfig<D = any> {
     blocking: boolean;
     withCloseButton: boolean;
+    data?: D;
 
-    constructor(data?: IModalConfig) {
-        this.blocking = !!data && typeof data.blocking === 'boolean' ? data.blocking : true;
-        this.withCloseButton = !!data && typeof data.withCloseButton === 'boolean' ? data.withCloseButton : false;
+    constructor(config?: IModalConfig<D>) {
+        this.blocking = !!config && typeof config.blocking === 'boolean' ? config.blocking : true;
+        this.withCloseButton = !!config && typeof config.withCloseButton === 'boolean' ? config.withCloseButton : false;
+        this.data = config?.data;
     }
 }
 
-export class ModalRef<T = any> {
+export class ModalRef<D = any, R = any> {
```

Comment:
out of curiosity: what value (other than avoiding an `any` type) does the `R` is bringing?
I feel like we don't use it anywhere else (I mean we don't need to define a type when creating or opening a dialog, right?)

Reply:
The types are being defaulted to any, thus are not required when creating.  However, if they were defined you get the following benefits:
1. In modal component, specifying D provides typing for the data
2. In the opening component, specifying R provides typing for the value returned to `onClosed`